destination.go

```diff
@@ -93,20 +88,35 @@ type Destination interface {
 	mustEmbedUnimplementedDestination()
 }
 
+// DestinationConfig represents the configuration containing all configuration
+// keys that a destination expects. The type needs to implement [Validatable],
+// which will be used to automatically validate the config when configuring the
+// connector.
+type DestinationConfig interface {
+	Validatable
+
+	mustEmbedUnimplementedDestinationConfig()
```

Comment:
Not in love with the naming, but can't think of anything better at the moment.

Reply:
True... I'll keep that in mind, if something which is more readable comes to my mind, I'll run it by you.